### DISCUSSION TITLE
Is it possible to use AegoCD's Dex to authenticate applications?

### DISCUSSION THREAD

**Discussion by quintesse** [argoproj/argo-cd#13594] [Q&A]
Just what the title says, given the fact that ArgoCD is already using Dex is it possible to use it for authenticating the apps that ArgoCD manages?

Or should I just do the opposite and set up my own Dex to handle the authentication of all my apps (including ArgoCD)?

**Comment by jaideepr97** [comment]
Hi @quintesse 
Sorry, maybe I don't fully understand your use case - could you elaborate on what you mean by authentication of apps? 

Are you perhaps deploying a serviceaccount as a part of your app resources that you want authenticated?

**Reply by quintesse** [reply]
Yeah, I think it's just me not having a clear idea of how things work and trying to using the wrong tool for the job.

I'll tell you what I'm looking for, but I'm already pretty sure it can't be done the way I thought/hoped.

What I'm looking for is a way to set up a SSO for any web applications running in the cluster. So instead of applications having to come up with their own authentication I just want to offer them a simple and declarative way of hooking into Dex. The same way you can set up TLS access for an ingress using cert-manager/letsencrypt I'm looking to do the same for authentication. Dex looks like the proper piece of the puzzle, but I'm still looking for the other pieces.

Anyway, thanks for replying. I'll close this question as "misguided" :-)
